### PR TITLE
MK-435 - implemented the backend services for comments

### PR DESCRIPTION
**Pull request by kazoompa**

### CODE REVIEW

File: mica-core/src/main/java/org/obiba/mica/core/domain/Comment.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+public class Comment extends AbstractAuditableDocument {
+
+  private static final long serialVersionUID = -1490617732167157048L;
+
+  private String author;
+
+  private String message;
+
+  private String classId;
```
Comment (ymarcon) [inline mica-core/src/main/java/org/obiba/mica/core/domain/Comment.java]: why not having 2 fields?

File: mica-rest/src/main/java/org/obiba/mica/access/rest/DataAccessRequestsResource.java [hunk 2]
```diff
@@ -72,7 +72,7 @@ public Response create(Mica.DataAccessRequestDto dto, @Context UriInfo uriInfo)
     dataAccessRequestService.save(request);
 
     subjectAclService.addPermission("/data-access-request", "VIEW,EDIT", request.getId());
-    subjectAclService.addPermission("/data-access-request/" + request.getId() + "/comments", "ADD");
+    subjectAclService.addPermission("/data-access-request/" + request.getId() + "/comments", "ADD,VIEW,EDIT");
```
Comment (ymarcon) [inline mica-rest/src/main/java/org/obiba/mica/access/rest/DataAccessRequestsResource.java]: VIEW and EDIT only apply to comment instances

File: mica-core/src/main/java/org/obiba/mica/core/domain/Comment.java (hunk 1 above)
Comment (kazoompa) [inline mica-core/src/main/java/org/obiba/mica/core/domain/Comment.java]: No earth shaking reason, merely for simplicity, can change to two.

File: mica-rest/src/main/java/org/obiba/mica/access/rest/DataAccessRequestsResource.java (hunk 2 above)
Comment (kazoompa) [inline mica-rest/src/main/java/org/obiba/mica/access/rest/DataAccessRequestsResource.java]: Needed an edit (deleting) privilege for deleting DAR associated comments, I will have to check permissions per comment instance now.

File: mica-rest/src/main/java/org/obiba/mica/access/rest/DataAccessRequestResource.java [hunk 3]
```diff
@@ -74,11 +85,62 @@ public Response delete(@PathParam("id") String id) {
     return Response.noContent().build();
   }
 
+  @GET
+  @Path("/comments")
+  public List<Mica.CommentDto> comments(@PathParam("id") String id) {
+    subjectAclService.checkPermission("/data-access-request/" + id + "/comments", "VIEW");
```
Comment (ymarcon) [inline mica-rest/src/main/java/org/obiba/mica/access/rest/DataAccessRequestResource.java]: subjectAclService.checkPermission("/data-access-request/", "VIEW", id);

File: mica-rest/src/main/java/org/obiba/mica/access/rest/DataAccessRequestResource.java [hunk 4]
```diff
@@ -74,11 +85,62 @@ public Response delete(@PathParam("id") String id) {
...
+
+    subjectAclService.addPermission("/data-access-request/" + id + "/comment", "VIEW,EDIT", comment.getId());
+
+    return Response.noContent().build();
+  }
+
+  @GET
+  @Path("/comment/{commentId}")
+  public Mica.CommentDto getComment(@PathParam("id") String id, @PathParam("commentId") String commentId) {
+    subjectAclService.checkPermission("/data-access-request/" + id + "/comment", "VIEW", commentId);
```
Comment (ymarcon) [inline mica-rest/src/main/java/org/obiba/mica/access/rest/DataAccessRequestResource.java]: subjectAclService.checkPermission("/data-access-request/", "VIEW", id);

**Pull request merged**